### ISSUE TITLE
jsonencode() does not convert values from &encoding

### ISSUE THREAD

**Issue by ZyX-I** [vim/vim#584]
Consider the following script:

```
vim -u NONE --cmd 'set encoding=cp1251 termencoding=utf8' --cmd 'let s="\xAB"' --cmd 'call writefile([jsonencode(s)], "/tmp/jetest.json", "b")' --cmd qa
cat /tmp/jetest.json | xxd
```

This will result in file with contents

```
00000000: 22ab 22                                  "."
```

, but JSON specification allows _only_ UTF-8. This should result in

```
00000000: 22c2 ab22                                ".."
```

because string should be converted from `&encoding` to UTF-8.

**Comment by brammool** [comment]
Nikolai Pavlov wrote:

> Consider the following script:
> 
> ```
> vim -u NONE --cmd 'set encoding=cp1251 termencoding=utf8' --cmd 'let s="\xAB"' --cmd 'call writefile([jsonencode(s)], "/tmp/jetest.json", "b")' --cmd qa
> cat /tmp/jetest.json | xxd
> ```
> 
> This will result in file with contents
> 
> ```
> 00000000: 22ab 22                                  "."
> ```
> 
> , but JSON specification allows _only_ UTF-8. This should result in
> 
> ```
> 00000000: 22c2 ab22                                ".."
> ```
> 
> because string should be converted from `&encoding` to UTF-8.

I'll put this one in the todo list.

## 

Due knot trussed yore spell chequer two fined awl miss steaks.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\
\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by Milly** [comment]
This issue is fixed at v7.4.1434 (b6ff811).

**Comment by brammool** [comment]
Closing then.
